File: Frontend/manga-world/src/Model/WishList.ts
import Manga from "./Manga";
import User from "./User";

class WishList {
  userId: number;
  mid: number;
  addedDate: Date;
  user: User;
  manga: Manga;

  constructor(
    userId: number,
    mid: number,
    addedDate: Date,
    user: User,
    manga: Manga
  ) {
    this.userId = userId;
    this.mid = mid;
    this.addedDate = addedDate;
    this.user = user;
    this.manga = manga;
  }
}

export default WishList;

File: Backend/manga-world/src/main/java/com/amakakeru/mangaworld/Entity/Wishlist.java
package com.amakakeru.mangaworld.Entity;

import com.amakakeru.mangaworld.Entity.IdClasses.WishlistId;
import jakarta.persistence.*;
import lombok.Data;

@Data
@Entity
@IdClass(WishlistId.class)
@Table(name = "wishlist")
public class Wishlist {

    @Id
    @Column(name = "user_id")
    private Long userId;

    @Id
    @Column(name = "m_id")
    private Long mid;

    @Column(name = "added_date", nullable = false)
    private java.sql.Timestamp addedDate;

    @ManyToOne
    @JoinColumn(name = "user_id", insertable = false, updatable = false)
    private User user;

    @ManyToOne
    @JoinColumn(name = "m_id", insertable = false, updatable = false)
    private Manga manga;
}
